Hangman/HangmanGame.py

```diff
@@ -0,0 +1,119 @@
+from Hangman import Hangman
+from WordGenerator import WordGenerator
+
+
+class HangmanGame:
+    def __init__(self) -> None:
+        self.total_points = 0
+        self.word_generator = WordGenerator()
+        self.hangman = Hangman()
+        self.word = ''
+        self.incorrect_guess_count = 0
+        self.word_display = []
+        self.guesses = []
+        self.status = ''
+
+    def start_game(self) -> None:
+        self.__set_word(self.word_generator.get_next_word())
+        self.__set_status()
+        self.word_display = ['_' for _ in self.word]
+        print("Welcome to hangman")
+        print(self.get_position())
+
+        game_over = (self.get_status() == 'won' or self.get_status() == 'lost')
+
+        # Take input till game is not over
+        while not game_over:
+            self.guess_letter()
+            print(self.get_hangman())
+            print(self.get_position())
+            game_over = (self.get_status() == 'won' or self.get_status() == 'lost')
+
+        if self.get_status() == 'won':
+            print('Congratulations! You guessed the word correctly.')
+        else:
+            print(f'The word is: {self.get_word()}')
+            print('I am sure you will do better next time. :)')
+
+        want_to_restart = (input("To restart, enter Yes: (case insensitive) ")).lower()
+        if want_to_restart == "yes":
+            self.__init__()
+            self.start_game()
+        print('\nThanks for playing!')
+
+    def __set_word(self, word: str) -> None:
+        self.word = word
+
+    def get_word(self) -> str:
+        return self.word
+
+    def get_hangman(self) -> str:
+        if self.get_status() == 'won':
+            return self.hangman.hangman_on_win()
+        else:
+            return self.hangman.get_hangman()
+
+    def get_position(self) -> str:
+        position_string = f'\t {" ".join(self.word_display)}\n'
+        if len(self.guesses) != 0:
+            position_string += f'Letters guessed: {", ".join(self.guesses)}'
+        
+        return position_string
+
+    def __set_status(self) -> None:
+        if self.incorrect_guess_count >= self.hangman.get_hangman_length():
+            self.status = 'lost'
+        elif self.word == ''.join(self.word_display):
+            self.status = 'won'
+        else:
+            self.status = 'guessing'
+
+    def get_status(self) -> str:
+        return self.status
+
+    def guess_letter(self) -> None:
+        guess = self.get_valid_guess()
+        self.guesses.append(guess)
+
+        if guess in self.word:
+            self.update_word_display(guess)
+        else:
+            self.incorrect_guess_count += 1
+            self.hangman.on_incorrect_guess()
+        self.__set_status()
```

Comment:
why __ ?

Reply:
Only HangmanGame class is using this set_status and any other class should not be able to change game status. So, it should be private right?